### PR TITLE
Add support for dynamic Pebble background colors

### PR DESCRIPTION
**Pull request by KazWolfe**
## Changes
- Add a couple additional icon types
  - Add Lighthouse (currently unused)
  - Add Transit (public transportation app)
- Tweak the colors on existing icon types
- Implement logic to grab primary (vibrant) color from app logo
  - The color will be used when displaying a notification for an app that does not have any configs bound to it.
- Alter NotificationType to support a color (named pebbleColor)
- Alter the Pebble notification poster to listen to the color from the notification
- Alter the DeviceCommunicationService to allow for color passthrough.
- Add logic to convert HEX or Integer representations of RGB888 colors to Pebble RGB222 format.

## Linked Issues
* #815: **Allow Automatic Notification Colors on Pebble**

## Reasoning For Commit
See issue above. This commit will add in support for dynamic background colors for Pebble notifications. It has been tested on live hardware with a Pebble Time and a Pebble Time Round, and certain aspects are covered by automated tests to ensure code quality.

There should be no (known) regressions to the system due to this code.

### CODE REVIEW

**Comment by danielegobbetti** [comment]
Great! Thanks, I will review ASAP.

**Comment by KazWolfe** [comment]
Added commit 0033cef to this PR.

## Changes

- Very simple patch to make the package name retrieved lowercase.

## Linked Issues
None

## Reason
Android packages tend to be lowercase, and we shouldn't care about case when matching. It doesn't seem too important, and I'm not aware of any apps where a case difference *actually* matters.

One such affected app, however, is Slack. The code is looking for packages named `com.slack` for Slack notifications, but the actual name of the app is `com.Slack`.

This case-insensitive behavior *appears* to be used by the official Pebble app, though really it's up to GB if we want it too. However, I don't see a reason not to.

**Comment by danielegobbetti** [comment]
BTW, perhaps you can take a look at #695 ? 😉

I mean the parts about handling the long push of the middle button on the pebble watch.

**Comment by aveao** [comment]
The issue wasn't linked but this is about #815, btw.

**Comment by danielegobbetti** [comment]
Tested and confirmed working, thanks a lot!

Before we merge, please run "Code" -> "Optimize imports" on your changed files and push the changes.

P.S. In order to test I had to comment the functionality that uses the hardcoded colors, as I don't have unsupported apps on my device :)

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/model/NotificationType.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
     GENERIC_EMAIL(PebbleIconID.GENERIC_EMAIL, PebbleColor.Orange),
     GENERIC_NAVIGATION(PebbleIconID.LOCATION, PebbleColor.Orange),
     GENERIC_PHONE(PebbleIconID.DURING_PHONE_CALL, PebbleColor.JaegerGreen),
-    GENERIC_SMS(PebbleIconID.GENERIC_SMS, PebbleColor.VividViolet),
+    GENERIC_SMS(PebbleIconID.GENERIC_SMS, PebbleColor.IslamicGreen),
```
Comment (danielegobbetti) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/model/NotificationType.java]: Why the change for SMS?

Comment (aveao) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/model/NotificationType.java]: Official App uses `IslamicGreen` for messages, not `VividViolet`.

Comment (KazWolfe) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/model/NotificationType.java]: Effectively what @aveao said.

I noted through testing that a value was unexpectedly violet, so I changed the SMS field. This is a follow-up of sorts to #816.

**Comment by KazWolfe** [comment]
@danielegobbetti I have a commit here standing by with IntelliJ's idea of import optimization. This is global, and is changing 226 files (none of which are already in this change).

If I remember correctly, my own IDE is set to auto-optimize imports of altered files.

If you still want me to push, let me know.

**Comment by danielegobbetti** [comment]
Thanks! I still don't get why the imports in DeviceCommunicationService are reordered like they are... Perhaps it's a different convention in android studio / intellij idea. 
If you want to try another round of optimization on that file, I'd appreciate that, otherwise I will merge manually tomorrow (I just chose code -> optimize imports earlier today when testing your code and all the imports were correct again, so I can do it really quickly)

**Comment by KazWolfe** [comment]
Re-running optimize imports on that file returns the same result.

I'm just using the generic [`AndroidStyle.xml`](https://github.com/android/platform_development/blob/master/ide/intellij/codestyles/AndroidStyle.xml), and this is what I'm getting.

If you want, I can just manually move the `import static`s down below the dynamic imports, as that's really the only big difference I can see.

**Comment by danielegobbetti** [comment]
I let optimize imports run and reverted the change of the SMS color (better preserve the continuity for Gadgetbridge users instead of official app newcomers).

**Comment by KazWolfe** [comment]
Would it be possible for me to get a screenshot or similar of your import config so I can properly configure my IDE? I'd rather not go through these hoops of trying (and failing to) optimize imports again.
